source/lib.ts

```diff
@@ -76,18 +76,23 @@ const promisifyStore = (passedStore: LegacyStore | Store): Store => {
 	return new PromisifiedStore()
 }
 
+type OptionsWithLegacyStore = Omit<Partial<Options>, 'store'> & {
+	store?: Store | LegacyStore
+}
+
 /**
  * Type-checks and adds the defaults for options the user has not specified.
  *
  * @param options {Options} - The options the user specifies.
  *
  * @returns {Options} - A complete configuration object.
  */
-const parseOptions = (
-	passedOptions: Omit<Partial<Options>, 'store'> & {
-		store?: Store | LegacyStore
-	},
-): Options => {
+const parseOptions = (passedOptions: OptionsWithLegacyStore): Options => {
+	// Passing undefined should be equivalent to not passing an option at all, so we'll
+	// omit all fields where their value is undefined.
+	const omittedOptions: OptionsWithLegacyStore =
+		omitUndefinedOptions(passedOptions)
```

Comment:
Maybe we could put the `omitUndefinedOptions` code here itself instead of declaring a new function? It is just a loop after all.

And the `ommittedOptions` variable could be of type `Partial<Options>` once we implement the change above.

Reply:
Also, could we rename `ommittedOptions` to something like `notUndefinedOptions`?

Reply:
Renamed the variable -- however, I moved the omitting function to a separate function because it's not very clean. It's only used once so it's fine to inline, but it's not great :s